### PR TITLE
Centralize Ruby Version to `.ruby-version`

### PR DESCRIPTION
**Pull request by alexcrocha**
## What are you trying to accomplish?
The `.ruby-version` file is the ecosystem standard for defining a Ruby version.  This PR adds the `.ruby-version` file, ensures a `required_ruby_version` is set, and removes all other references to Ruby in this repository, aligning it with the standard.

Verify that the changes in the PR meets the following requirements or adjust manually to make it compliant:
  - [x] `.ruby-version` file is present with the correct Ruby version defined
  - [x] A `required_ruby_version` in your gemspec is set
  - [x] There is no Ruby version present in the  `dev.yml` Ruby task (before: `- ruby: x.x.x`, after: `- ruby`)
  - [x] There is no Ruby version/requirement referenced in the `Gemfile` (no lines with `ruby  <some-version>`)
  - [x] A `Gemfile.lock` is built with the defined Ruby version
  - [x] The version of Rubocop installed is 1.61.0 or greater
  - [x] There is no `TargetRubyVersion` defined  in `rubocop.yml` 
  - [x] There is no Ruby argument  present in  `ruby/setup-ruby` Github Actions that do **not**  run on a Ruby matrix (no lines with `ruby-version: “x.x”`)

This PR will be merged if there isn't any activity after 4 weeks.

### CODE REVIEW

**Comment by kmcphillips** [comment]
The same changes will need to be made here:
https://github.com/Shopify/measured-rails/blob/main/dev.yml

We can release them together. (actually we _must_ release them together)

**Comment by alexcrocha** [comment]
> The same changes will need to be made here: https://github.com/Shopify/measured-rails/blob/main/dev.yml
> 
> We can release them together. (actually we _must_ release them together)

Thank you for the heads up 👍 It's on our list and I will add that note.

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.0.2
```
[the remark this answers is not in the document]
Comment (alexcrocha) [reply .ruby-version:1]: The Ruby version being used for CI is defined a few lines below and is referencing the CI matrix:
https://github.com/Shopify/measured/blob/a1c1758868750f64234241f560bb8d2bd894ba6f/.github/workflows/ci.yml#L27-L28

You can verify that the CI runs and passes successfully for all Ruby versions specified in the matrix by checking the results here: https://github.com/Shopify/measured/actions/runs/8300926380 🎉

[the remark this answers is not in the document]
Comment (alexcrocha) [reply .ruby-version:1]: (It's all good - you're making me strengthen my understanding as well)

The CI will only look at the `.ruby-version` file when the `ruby-version` input is not specified. Thankfully, this is the behaviour we want here as we want CI to test against other versions as well.

If it helps, I am referencing these [docs](https://github.com/ruby/setup-ruby?tab=readme-ov-file#usage).

[the remark this answers is not in the document]
Comment (kmcphillips) [reply .ruby-version:1]: Amazing. Thank you!

**Pull request merged**